prql-compiler/src/lib.rs

```diff
@@ -143,6 +150,23 @@ impl Target {
     }
 }
 
+impl FromStr for Target {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Target, Self::Err> {
+        if s.starts_with("sql.") {
+            let maybe_dialect = s.strip_prefix("sql.").unwrap_or(s);
+            let dialect = sql::Dialect::from_str(maybe_dialect).unwrap_or_default();
+            Ok(Target::Sql(Some(dialect)))
+        } else {
+            Err(Error::new(Reason::NotFound {
+                name: format!("{s:?}"),
+                namespace: "target".to_string(),
+            }))
+        }
```

Comment:
It seems like we have two identical logic branches here. Could we do something like:

```suggestion
            s.strip_prefix("sql.")
                      .map(|d| Target::Sql(Some(d)))
                      .map_err(
                         Err(Error::new(Reason::NotFound {
                         name: format!("{s:?}"),
                         namespace: "target".to_string(),
            )?
```

(I may be missing something, and I imagine this won't quite be correct, but I _think_ it should be close)

Reply:
Thank you for reviewing.
My intention was to branch the process for each prefix, but so far it is unnecessary.

Reply:
I think it got better thanks to your advice, but after a lot of trial and error the if statement remained. (I couldn't pass the tests otherwise)

Reply:
(just saw this — yes I was a bit optimistic! Thanks for making it better!)